BayesianEstimation/AdaptiveMetropolisWithinGibbs.cs

```diff
@@ -0,0 +1,285 @@
+    /// normal distribution, but with fatter tails; this method is more tolerant of outliers than fitting a normal distribution). 
+    /// The mu, sigma, and nu parameters of the t-distribution are estimated by sampling from normal, uniform, and exponential distributions, 
+    /// respectively, using a Markov Chain Monte Carlo (MCMC) method and calculating their probability of describing the data (Bayesian statistics). 
+    /// Eventually, after enough iterations, the algorithm hopefully converges on an accurate answer, with some amount of measureable 
+    /// uncertainty for each parameter.
+    /// 
+    /// You can fit one-sample or two-sample data by making the "data2" parameter in the constructor null or not null, respectively.
+    /// 
+    /// If your data do not conform to a t-distribution, you can inherit this class and then define a new PosteriorProbability
+    /// function. The rest of the algorithm remains the same (the Metropolis within Gibbs sampler samples from the distributions 
+    /// you define in order to fit the distribution you want to your data). See FlashLfqBayesianEstimator.cs for an example of defining
+    /// a different posterior probability function.
+    /// 
+    /// See:
+    /// Bayesian Estimation Supersedes the t Test. Kruschke, J. K., Journal of Experimental Psychology, 2013.
+    /// 
+    /// This code is adapted from Javascript code taken from http://sumsar.net/best_online/ . This webpage is a useful
+    /// interactive example of J. Kruschke's Bayesian equivalent of a t-test.
+    /// </summary>
+    public class AdaptiveMetropolisWithinGibbs
+    {
+        public List<double[]> markovChain { get; private set; }
+        private double[] logSd;
+        private double[] currentState;
+        private double[] acceptanceCount;
+        private int numParams;
+        private int batchCount;
+        private int batchSize;
+        private MersenneTwister random;
+        private List<double>[] Data;
+        private bool isTwoSample;
+        private List<Tuple<double, double>>[] acceptableParameterRanges;
+
+        // these are the parameters of the prior probability distributions
+        private double meanMu; // mean of prior prob mean normal distribution
+        private double sdMu; // sd of prior prob mean normal distribution
+        private double sigmaLow; // lower bound of prior prob sd uniform distribution
+        private double sigmaHigh; // upper bound of prior prob sd uniform distribution
+        private double exp; // exponent of prior prob exponential distribution
+
+        /// <summary>
+        /// Construct the adaptive Metropolis within Gibbs sampler. Leave data2 null for one-sample data.
+        /// </summary>
+        public AdaptiveMetropolisWithinGibbs(List<double> data1, List<double> data2, int batch_size = 3, int? seed = null)
+        {
+            if (seed != null)
+            {
+                random = new MersenneTwister(seed.Value, true);
```

Comment:
don't need to change anything. might be good to have a "good" generic seed generator in mzlib that was accessible to each method that needed one. allows for more consistency.